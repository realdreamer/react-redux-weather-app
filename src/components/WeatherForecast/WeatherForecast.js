import React from 'react';
import DayWeatherInfo from '../DayWeatherInfo/DayWeatherInfo';

import './scss/WeatherForecast.css';

const WeatherForecast = ({weather}) => {
  if (weather) {
    return (
      <section className="weather-info">
        { weather.map( (weatherDetail) => {
            return <DayWeatherInfo key={weatherDetail.id} info={weatherDetail} />;
          })
        }
      </section>
    )
  }
  else {
    return null;
  }
}

export default WeatherForecast;
