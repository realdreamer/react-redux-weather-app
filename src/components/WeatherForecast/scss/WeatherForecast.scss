.weather-info {
  display: flex;
  flex-direction: column;
  min-height: calc(100% - 41px);
}
