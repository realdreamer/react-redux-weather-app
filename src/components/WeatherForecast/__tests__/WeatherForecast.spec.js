import React from 'react';
import renderer from 'react-test-renderer';
import WeatherForecast from '../WeatherForecast';

describe('Weather Info', () => {
  it('render weather info component without fail', () => {
    const Component = renderer.create(<WeatherForecast />).toJSON();
    expect(Component).toMatchSnapshot();
  });

  it('render weather info component by passing props in', () => {
    const props = [{ id: 5163259682357248, weather_state_name: "Light Cloud", weather_state_abbr: "lc", wind_direction_compass: "SSE", created: "2018-05-20T10:28:43.158520Z", applicable_date: "2018-05-20", min_temp: 29.2175, max_temp: 35.44, the_temp: 37, wind_speed: 7.957708449322623, wind_direction: 164.7817856137123, air_pressure: 1019.01, humidity: 71, visibility: 4.05755388530979, predictability: 70 }];
    const Component = renderer.create(<WeatherForecast weather={props} />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
