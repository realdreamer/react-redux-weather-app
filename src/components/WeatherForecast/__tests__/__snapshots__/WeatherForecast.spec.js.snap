// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Info render weather info component by passing props in 1`] = `
<section
  className="weather-info"
>
  <section
    className="forecast"
  >
    <div
      className="forecast-info"
    >
      <h4
        className="forecast__date"
      >
        2018-05-20
      </h4>
      <h2
        className="forecast__state"
      >
        Light Cloud
      </h2>
    </div>
    <div
      className="forecaste-state"
    >
      <i
        className="wi wi-cloud"
      />
    </div>
  </section>
</section>
`;

exports[`Weather Info render weather info component without fail 1`] = `null`;
